problem2.py

```diff
@@ -0,0 +1,37 @@
+#timecomplexity O()??
+#spacecomplexity O()??
```

Comment:
Both time & space here will be O(n * m), where n is the length of coins array and m is the amount.